src/main/java/com/minecolonies/coremod/entity/ai/citizen/lumberjack/EntityAIWorkLumberjack.java

```diff
@@ -329,28 +329,43 @@ private IAIState findTree()
 
         if (pathResult == null || pathResult.treeLocation == null)
         {
-
             final BuildingLumberjack buildingLumberjack = (BuildingLumberjack) building;
             final Map<String, List<ItemStorage>> copy = buildingLumberjack.getCopyOfAllowedItems();
             if (buildingLumberjack.shouldRestrict())
             {
                 final BlockPos startPos = buildingLumberjack.getStartRestriction();
                 final BlockPos endPos = buildingLumberjack.getEndRestriction();
 
-                pathResult = worker.getNavigator().moveToTree(
-                  startPos, endPos,
-                  1.0D,
-                  copy.getOrDefault(SAPLINGS_LIST, Collections.emptyList()),
-                  worker.getCitizenColonyHandler().getColony()
-                );
+                pathResult = worker.getNavigator()
+                    .moveToTree(startPos,
+                        endPos,
+                        1.0D,
+                        copy.getOrDefault(SAPLINGS_LIST, Collections.emptyList()),
+                        worker.getCitizenColonyHandler().getColony());
+                
+                if (pathResult.treeLocation == null)
+                {
+                    return LUMBERJACK_NO_TREES_FOUND;
+                }
             }
             else
             {
                 pathResult = worker.getNavigator()
-                               .moveToTree(SEARCH_RANGE + searchIncrement,
-                                 1.0D,
-                                 copy.getOrDefault(SAPLINGS_LIST, Collections.emptyList()),
-                                 worker.getCitizenColonyHandler().getColony());
+                    .moveToTree(SEARCH_RANGE + searchIncrement,
+                        1.0D,
+                        copy.getOrDefault(SAPLINGS_LIST, Collections.emptyList()),
+                        worker.getCitizenColonyHandler().getColony());
+
+                if (pathResult.treeLocation == null)
```

Comment:
we have to do a check if the path finished computing here

Reply:
It's just a draft, I would like to replace entire after-search or exactly not sure how and what to do here